lib/core/polyfills/index.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * Note:
+ * This file is run via browserify to pull in the required dependencies.
+ * See - './build/polyfills-generator'
+ */
+
+/**
+ * Polyfill `Promise`
+ * Reference: https://www.npmjs.com/package/es6-promise
+ */
+if (!('Promise' in window)) {
+	require('es6-promise').polyfill();
+}
+
+/**
+ * Polyfill required TypedArray and functions
+ * Reference https://github.com/zloirock/core-js/
+ */
+if (!('Uint32Array' in window)) {
+	require('core-js/features/typed-array/uint32-array');
+}
+if (window.Uint32Array) {
+	if (!('some' in window.Uint32Array.prototype)) {
+		require('core-js/features/typed-array/some');
+	}
+	if (!('reduce' in window.Uint32Array.prototype)) {
+		require('core-js/features/typed-array/reduce');
+	}
+}
+
+/**
+ * Polyfill `WeakMap`
+ * Reference: https://github.com/polygonplanet/weakmap-polyfill
+ */
+require('weakmap-polyfill');
+
+/* eslint-disable */
+/*
+ These pollyfills came directly from the ES Specification it's self
+ Contained within:
+  - Object.assign
+  - Array.prototype.find
+*/
+if (typeof Object.assign !== 'function') {
```

Comment:
another thought is, given we have `core-js` already above for `reduce`, can we remove the copy pasted polyfills below and swap them out to features from `core-js`

Eg: `core-js(-pure)/es|stable|features/object/assign`

Reply:
I agree here, but don't think that needs to change for this PR to land